rotkehlchen/balances/historical.py

```diff
@@ -99,63 +265,230 @@ def get_assets_amounts(
-                direction == EventDirection.NEUTRAL
-            ):
-                continue
+            if (event_id := event.update_balances(current_balances)) is not None:
+                negative_balance_event = event_id
+                break
+
+            for asset in assets:
+                amounts[event.timestamp] += current_balances[asset]
+
+        return amounts, negative_balance_event
+
+    def get_historical_netvalue(
+            self,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> tuple[dict[Timestamp, FVal], list[tuple[str, Timestamp]], str | None]:
+        """Calculates historical net worth per day within the given time range.
 
-            if direction == EventDirection.IN:
-                total_amount += event.balance.amount
-            else:  # EventDirection.OUT
-                total_amount -= event.balance.amount
+        Uses asset balances per day combined with historical prices to calculate
+        the total net worth in the user's profit currency for each day. Stops calculation
+        if negative balance would be created for any asset.
+
+        May raise:
+            - NotFoundError if no events exist in the specified time period.
 
-            amounts[ts_ms_to_sec(event.timestamp)] = total_amount
-            if total_amount < ZERO:
-                last_event_id = event.event_identifier
+        Returns:
+            - A mapping of timestamps to net worth in main currency for each day
+            - A list of (asset id, timestamp) tuples where price data was missing
+            - The event_identifier that caused a negative balance if any, else None
+        """
+        events, main_currency = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts)
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found within {from_ts=} and {to_ts=}.')
+
+        negative_balance_event = None
+        current_balances: dict[Asset, FVal] = defaultdict(FVal)
+        daily_balances: dict[Timestamp, dict[Asset, FVal]] = {}
+        current_day = timestamp_to_daystart_timestamp(events[0].timestamp)
+        for event in events:
+            if (day_ts := timestamp_to_daystart_timestamp(event.timestamp)) > current_day:
+                daily_balances[current_day] = current_balances.copy()
+                current_day = day_ts
+
+            if (negative_balance_id := event.update_balances(current_balances)) is not None:
+                negative_balance_event = negative_balance_id
                 break
 
-        return amounts, last_event_id
+        if current_day is not None and negative_balance_event is None:
+            daily_balances[current_day] = current_balances.copy()
+
+        net_worth_per_day: dict[Timestamp, FVal] = {}
+        missing_price_points: list[tuple[str, Timestamp]] = []
+        for day_ts, asset_balances in daily_balances.items():
+            assets_to_price = [
```

Comment:
Why are you calling this `assets_to_prices`? This is just a list of assets. What name were you trying to give here?

Reply:
> Why are you calling this assets_to_prices? This is just a list of assets. What name were you trying to give here?

yeah, my bad. renamed it to `assets`